tests/Tests.Nybus/Filters/DeadLetterQueueErrorFilterTests.cs

```diff
@@ -0,0 +1,276 @@
+                TErrorDelegate errorDelegate,
+                TestException exception)
+            {
+                SetEntryAssembly(Assembly.GetCallingAssembly());
+
+                await HandleErrorAsync(sut, context, exception, errorDelegate);
+
+                Mock.Get(engine)
+                    .Verify(
+                        e => e.SendMessageToErrorQueueAsync(It.Is<Message>(m =>
+                            m.Headers["DLQ-Error-Assembly"].Equals(Assembly.GetEntryAssembly().GetName().Name))), Times.Once);
+
+                SetEntryAssembly(null);
+            }
+
+            private void SetEntryAssembly(Assembly assembly)
+            {
+#if NET472
+                AppDomainManager manager = new AppDomainManager();
+                FieldInfo entryAssemblyfield = manager.GetType().GetField("m_entryAssembly", BindingFlags.Instance | BindingFlags.NonPublic);
+                entryAssemblyfield.SetValue(manager, assembly);
+
+                AppDomain domain = AppDomain.CurrentDomain;
+                FieldInfo domainManagerField = domain.GetType().GetField("_domainManager", BindingFlags.Instance | BindingFlags.NonPublic);
+                domainManagerField.SetValue(domain, manager);
+#endif
+            }
+
+#if NET472
+            [Test, CustomAutoMoqData]
+            public async Task HandleErrorAsync_does_not_add_assemblyName_header_if_has_not_EntryAssembly(
+                [Frozen] IBusEngine engine,
+                DeadLetterQueueErrorFilter sut,
+                TContext context,
+                TErrorDelegate errorDelegate,
+                TestException exception)
+            {
+                await HandleErrorAsync(sut, context, exception, errorDelegate);
+
+                Mock.Get(engine)
+                    .Verify(
+                        e => e.SendMessageToErrorQueueAsync(It.Is<Message>(m =>
+                            !m.Headers.ContainsKey("DLQ-Error-Assembly"))), Times.Once);
+            }
+#endif
+
+            [Test, CustomAutoMoqData]
+            public async Task HandleErrorAsync_logs_error_on_exception(
+                [Frozen] IBusEngine engine,
+                [Frozen] ILogger<DeadLetterQueueErrorFilter> logger,
+                DeadLetterQueueErrorFilter sut,
+                TContext context,
+                TErrorDelegate errorDelegate,
+                TestException exception,
+                TestException engineException)
+            {
+                Mock.Get(engine).Setup(e => e.SendMessageToErrorQueueAsync(context.Message)).Throws(engineException);
+
+                Mock.Get(logger).Setup(
+                    l => l.Log(It.IsAny<LogLevel>(), It.IsAny<EventId>(),
+                        It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(),
+                        It.IsAny<Func<It.IsAnyType, Exception, string>>())).Verifiable();
+
+                await HandleErrorAsync(sut, context, exception, errorDelegate);
+
+                Mock.Get(logger).Verify(
```

Comment:
This should work too.

```suggestion
                Mock.Get(logger).Verify(
                    l => l.Log(LogLevel.Error, It.IsAny<EventId>(),
                        It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(),
                        It.IsAny<Func<It.IsAnyType, Exception, string>>()), Times.Once);
```

Reply:
Yes, good point